proposals/0035-linalg-matrix.md

```diff
@@ -1119,20 +1161,50 @@ Multiply(const Matrix<T, M, K, MatrixUse::A, MatrixScope::Wave>,
 
 template <typename OutputElTy, bool MatrixTranspose, typename InputElTy, uint M,
           uint K, MatrixComponentType MatrixDT, MatrixScope Scope>
-vector<OutputElTy, K>
-Multiply(vector<InputElTy, M> InputVector,
-         Matrix<MatrixDT, M, K, MatrixUse::B, Scope> MatB);
+vector<OutputElTy, K> Multiply(vector<InputElTy, M>,
+                               Matrix<MatrixDT, M, K, MatrixUse::B, Scope>);
 
 template <typename OutputElTy, typename InputElTy, typename BiasElTy, uint M,
           uint K, MatrixComponentType MatrixDT, MatrixScope Scope,
           bool MatrixTranspose>
-vector<OutputElTy, K>
-MultiplyAdd(vector<InputElTy, M> InputVector,
-            Matrix<MatrixDT, M, K, MatrixUse::B, Scope> MatB,
-            vector<BiasElTy, K> BiasVector);
+vector<OutputElTy, K> MultiplyAdd(vector<InputElTy, M>,
+                                  Matrix<MatrixDT, M, K, MatrixUse::B, Scope>,
+                                  vector<BiasElTy, K>);
 
 } // namespace linalg
 } // namespace dx
+
+RWByteAddressBuffer B : register(u0);
+
+void WaveMatrixExample() {
+  using namespace dx::linalg;
+  using MatrixATy =
+      Matrix<MatrixComponentType::F16, 8, 32, MatrixUse::A, MatrixScope::Wave>;
+  using MatrixBTy =
+      Matrix<MatrixComponentType::F16, 32, 16, MatrixUse::B, MatrixScope::Wave>;
+  using MatrixAccumTy = Matrix<MatrixComponentType::F16, 8, 16,
+                               MatrixUse::Accumulator, MatrixScope::Wave>;
+  using MatrixAccum32Ty = Matrix<MatrixComponentType::F32, 8, 16,
+                                 MatrixUse::Accumulator, MatrixScope::Wave>;
+
+  MatrixATy MatA = Matrix<MatrixComponentType::F16, 8, 32, MatrixUse::A,
+                          MatrixScope::Wave>::Load(B, 0, 8 * 4, false);
+  MatrixBTy MatB = Matrix<MatrixComponentType::F16, 32, 16, MatrixUse::B,
+                          MatrixScope::Wave>::Load(B, 0, 32 * 4, false);
+  MatrixAccumTy Accum = Multiply(MatA, MatB);
+  MatrixAccum32Ty Accum32 = Multiply<MatrixComponentType::F32>(MatA, MatB);
+}
+
+void CoopVec() {
+  using namespace dx::linalg;
+  using MatrixBTy =
+      Matrix<MatrixComponentType::F16, 32, 16, MatrixUse::B, MatrixScope::Wave>;
+
+  vector<float16_t, 32> Vec = (vector<float16_t, 32>)0;
+  MatrixBTy MatB = MatrixBTy::Load(B, 0, 32 * 4, false);
+  vector<float16_t, 16> Accum =
+      Multiply<float16_t, /*transpose*/ false>(Vec, MatB);

```

Comment:
nit: Sorry I should have said this sooner but it just occured to me.  If the proper thing is to always put a `/*transpose*/` comment, we should consider an explicitly named enum

Names could be improved but here is a potential example
```c++
enum MatrixTranspose  {
   None,
   Transpose
};

vector<float16_t, 16> Accum =
      Multiply<float16_t, MatrixTranspose::None>(Vec, MatB);
```

Reply:
I think I can actually remove the Transpose argument. So I'll do that.